### PR TITLE
[SAIC-760] Show the global progress of the image migration

### PR DESCRIPTION
**Pull request by SVilgelm**
Show information about total number of images to be migrated, total size,
progress of migration.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:159: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:240: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:281: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:292: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:295: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:328: [W0621(redefined-outer-name), GlanceImage.deploy] Redefining name 'image' from outer scope (line 31)
cloudferrylib/os/image/glance_image.py:305: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/image/glance_image.py:385: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:391: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:400: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:403: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:423: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:450: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:483: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions and pass the % parameters as arguments

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:160: [I0011(locally-disabled), ] Locally disabling protected-access (W0212)
cloudferrylib/os/image/glance_image.py:241: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:304: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils.test_sizeof_format
tests/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:159: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:240: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:303: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1753/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils.test_sizeof_format
tests/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:159: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:240: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:303: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.utils.test_sizeof_format
tests/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:159: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:240: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:303: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.utils.test_sizeof_format
tests/cloudferrylib/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:159: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:240: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:303: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -320,13 +319,61 @@ def deploy(self, info):
...
+        LOG.info('Total number of images to be migrated: %d, '
+                 'total size: %s\n'
+                 'Number of private images: %d\n'
+                 'Number of public images: %d\n'
+                 'Number of already migrated images: %d, total size: %s',
+                 num_total_images, total_size_hr,
+                 num_private, num_public, num_migrated, migrated_size_hr)
+        LOG.info('List of private images:\n%s', '\n'.join(list_private))
+        LOG.info('List of public images:\n%s', '\n'.join(list_public))
+        LOG.info('List of migrated images:\n%s', '\n'.join(list_migrated))
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: Let's move all that logic outside of `deploy()` method, `deploy()` is already way too complex

Comment (SVilgelm) [inline cloudferrylib/os/image/glance_image.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.utils.test_sizeof_format
tests/cloudferrylib/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:218: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:299: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:362: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.utils.test_sizeof_format
tests/cloudferrylib/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:218: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:299: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:362: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.utils.test_sizeof_format
tests/cloudferrylib/utils/test_sizeof_format.py:20: [E1002(super-on-old-class), SizeOfFormatTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:218: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:299: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:362: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
+1 tested in staging

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**